src/Npgsql/Util/PGUtil.cs

```diff
@@ -169,25 +169,31 @@ internal void CheckAndApply(NpgsqlConnector connector)
                 return;
 
             var timeLeft = TimeLeft;
-            if (timeLeft > TimeSpan.Zero)
-            {
-                // Set the remaining timeout on the read and write buffers
-                connector.ReadBuffer.Timeout = connector.WriteBuffer.Timeout = timeLeft;
-
-                // Note that we set UserTimeout as well, otherwise the read timeout will get overwritten in ReadMessage
-                // Note also that we must set the read buffer's timeout directly (above), since the SSL handshake
-                // reads data directly from the buffer, without going through ReadMessage.
-                connector.UserTimeout = (int)Math.Ceiling(timeLeft.TotalMilliseconds);
-            }
+            // Set the remaining timeout on the read and write buffers
+            connector.ReadBuffer.Timeout = connector.WriteBuffer.Timeout = timeLeft;
 
-            Check();
+            // Note that we set UserTimeout as well, otherwise the read timeout will get overwritten in ReadMessage
+            // Note also that we must set the read buffer's timeout directly (above), since the SSL handshake
+            // reads data directly from the buffer, without going through ReadMessage.
+            connector.UserTimeout = (int) Math.Ceiling(timeLeft.TotalMilliseconds);
         }
 
         internal bool IsSet => _expiration != DateTime.MaxValue;
 
         internal bool HasExpired => DateTime.UtcNow >= Expiration;
 
-        internal TimeSpan TimeLeft => IsSet ? Expiration - DateTime.UtcNow : Timeout.InfiniteTimeSpan;
+        internal TimeSpan TimeLeft
```

Comment:
nit: this is completely fine, but it doesn't feel quite right for this to be a property, since it now performs a check, may throw, etc. It feels like this should be a method, maybe something like CheckAndGetTimeLeft..

Reply:
Yeah, that makes sense.